src/main/java/core/basesyntax/strategy/handler/BalanceHandler.java

```diff
@@ -0,0 +1,16 @@
+package core.basesyntax.strategy.handler;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.Fruit;
+import core.basesyntax.service.OperationHandler;
+
+public class BalanceHandler implements OperationHandler {
+    @Override
+    public void apply(Fruit fruit, int quantity) {
+        if (quantity < 0) {
+            throw new RuntimeException("Balance cannot be negative");
```

Comment:
It's good that you're checking for negative quantities, but the error message could be more informative by including the actual quantity attempted to be set.